### PR TITLE
VCB-242 Fix alignment of data instructions

### PR DESCRIPTION
**Pull request by clooudch**
This snippet now works as expected:

```
      AREA |.data|, DATA, READONLY
byte  DCB 0x11
halfword
      DCW 0x2222
word  DCD 0x33333333

      AREA |.text|, CODE, READONLY

      ; Read byte variable
      LDR R0, =byte
      LDRB R1, [R0]

      ; Read halfword variable
      LDR R0, =halfword
      LDRH R1, [R0]

      ; Read word variable
      LDR R0, =word
      LDR R1, [R0]
```

In the previous version, the `DCW` instruction was properly aligned by the encoder, but the label `halfword` was not aligned correctly (it was pointing to alignment bytes between `DCB` and `DCW`). Therefor, I changed the code to add the label and the line mapping after the alignment.

### CODE REVIEW

**Comment by m-dot-wigger** [comment]
@clooudch there is one remaining TODO in parser.test.ts. I think this can be deleted what do you think?

File: src/assembler/encoder.ts [hunk 1]
```diff
@@ -279,12 +255,16 @@ function writeDataInstruction(
...
+    writer.mapLine(instruction.line)
+    writer.addLabel(instruction.label, instruction.line)
     for (const option of instruction.options) {
       writer.addDataRelocation(option.trim(), bytes.length, instruction.line)
       writer.writeBytes(bytes)
     }
     return
   } else if (instruction.name === 'ALIGN') {
+    writer.mapLine(instruction.line)
+    writer.addLabel(instruction.label, instruction.line)
```
Comment (m-dot-wigger) [inline src/assembler/encoder.ts:267]: can ALIGN have a label?

Comment (m-dot-wigger) [reply src/assembler/encoder.ts:267]: additionally everywhere else you aligned first and then called mapLine and addLabel. Is this intentional?

**Review by m-dot-wigger — CHANGES_REQUESTED** [review]
Please add test to check offset in opcode as in example code of this PR

File: src/assembler/encoder.ts (hunk 1 above)
Comment (clooudch) [reply src/assembler/encoder.ts]: You are right. It does not make sense to add a label or line mapping for a `ALIGN` instruction. I therefor removed it from the code.

**Review by m-dot-wigger — APPROVED** [review]
all looks good for me

**Pull request merged**